src/lib.rs

```diff
@@ -66,23 +66,43 @@ pub struct Module {
 }
 
 pub enum ModuleInner {
-  V1(Arc<Vec<u8>>),
+  V1(EszipV1),
   V2(EszipV2),
 }
 
 impl Module {
-  pub async fn source(&self) -> Arc<Vec<u8>> {
+  pub async fn source(&self) -> Option<Arc<Vec<u8>>> {
     match &self.inner {
-      ModuleInner::V1(source) => source.clone(),
-      ModuleInner::V2(eszip) => eszip.get_module_source(&self.specifier).await,
+      ModuleInner::V1(eszip_v1) => eszip_v1.get_module_source(&self.specifier),
+      ModuleInner::V2(eszip_v2) => {
+        eszip_v2.get_module_source(&self.specifier).await
+      }
+    }
+  }
+
+  pub async fn take_source(&self) -> Option<Arc<Vec<u8>>> {
```

Comment:
It would be good to have docstrings on both `source` and `take_source` to highlight the differences in behavior (and same for `source_map` and `take_source_map`).

Reply:
Updated :)